nixos/default.nix

```diff
@@ -129,9 +129,19 @@
         default = null;
       };
 
+      port = mkOption {
```

Comment:
Similar to the default values I commented on, this should be `urlParts.port` (and the description should read "The port to forward requests to").

Also, add the corresponding fields `urlParts.host` and `urlParts.protocol`.

Reply:
I may not fully understand `urlParts` but what would the purpose of `urlParts.port` be, wouldn't that prevent setting the port for each service individually?